events/tests/test_api.py

```diff
@@ -547,3 +576,77 @@ def test_staff_publish_event(snapshot, staff_api_client, event):
         PUBLISH_EVENT_MUTATION, variables=event_variables
     )
     assert "Event is already published" in str(executed["errors"])
+
+
+def test_enrol_occurrence(api_client, user_api_client, snapshot, occurrence):
+    non_authen_executed = api_client.execute(
+        ENROL_OCCURRENCE_MUTATION, variables=ENROL_OCCURRENCE_VARIABLES
+    )
+    assert_permission_denied(non_authen_executed)
+
+    child = ChildWithGuardianFactory(relationship__guardian__user=user_api_client.user)
+
+    enrolment_variables = deepcopy(ENROL_OCCURRENCE_VARIABLES)
+    enrolment_variables["input"]["occurrenceId"] = to_global_id(
+        "OccurrenceNode", occurrence.id
+    )
+    enrolment_variables["input"]["childId"] = to_global_id("ChildNode", child.id)
+
+    executed = user_api_client.execute(
+        ENROL_OCCURRENCE_MUTATION, variables=enrolment_variables
+    )
+    snapshot.assert_match(executed)
+
+    executed = user_api_client.execute(
+        ENROL_OCCURRENCE_MUTATION, variables=enrolment_variables
+    )
+
+    assert "Child already joined this event" in str(executed["errors"])
+
+    random_child = ChildWithGuardianFactory()
+    enrolment_variables["input"]["childId"] = to_global_id("ChildNode", random_child.id)
+
+    executed = user_api_client.execute(
+        ENROL_OCCURRENCE_MUTATION, variables=enrolment_variables
+    )
+    assert "does not exist" in str(executed["errors"])
```

Comment:
This is borderline bike-shedding, but it would be a bit nicer if this one big test that tests multiple things was divided into smaller tests, in this case I think there could be three different tests: one for a successful enrolment, one testing that you cannot enrol twice, and one testing that you cannot enrol someone else's child. Benefits: the tests would be easier to reason with and not dependent on each other.

Also as a future hint, there is `guardian_api_client` which might should be a bit easier to use than `user_api_client` for these tests that involve a guardian, of course no reason to change this stuff that is already implemented.

Reply:
Okay I'll split it to smaller tests. I just thought it'd be nicer to test the whole flow without repeating duplicated data simulation